common/src/stack/command/stack/commands/add/vm/storage/__init__.py

```diff
@@ -96,89 +105,52 @@ def run(self, params, args):
-
-				# First test for a gzip archive and replace the suffix for
-				# the uncompressed disk image name
-				if '.gz' in disk_path.suffix:
-
-					# Remove the archive extension
-					image_name = disk_path.name.replace(disk_path.suffix, '')
-					image_archive = str(disk_path)
-					images.append(image_name)
 
+					# Get the next disk name but only consider other disk type images
+					vol_file = vol_names[-1].replace(Path(vol_names[-1]).suffix, '')
+					vol_id = int(vol_file.split(vol_name)[1]) + 1
 				else:
-
-					# Otherwise assume the disk is an uncompressed
-					# image
-					if disk_path.suffix in image_formats:
-						images.append(disk)
-					else:
-						raise ParamError(self, 'Virtual Machine disk {disk} not a qcow2 or raw image')
+					vol_id += 1
+				image = f'{vol_name}{vol_id}.qcow2'
 
 			elif '/dev/' in disk:
 
 				# Unlike the other types of storage, mountpoints do not require
 				# a storage directory as it is just a mounted disk to a VM
 				disk_type = 'mountpoint'
 				mount_disk = disk
-				images.append('')
 
-			# For an archive containing multiple disk images
-			# Create a new entry in the table for each one
+			else:
+				if not disk_loc:
+					raise ParamError(self, 'storage_pool', 'needed for defined disks or images')
+
+				# Otherwise assume the disk is an premade image
+				disk_type = 'image'
+				image = str(disk_path)
+				if not self.valid_image(disk_path, image_formats):
+					raise ParamError(self, 'disks', f'Virtual Machine disk {disk} not found or not a qcow2 or raw image')
+
+			# Calculate the next device name (sdb, sdc, sdd...etc)
+			if disk_names:
+				disk_name = f'sd{chr(ord(disk_names[-1][-1]) + 1)}'
+
 			# Insert disks into the database
-			for image in images:
-
-				# Calculate the next device name (sda, sdb, sdc...etc)
-				if disk_names:
-					disk_name = f'sd{chr(ord(disk_names[-1][-1]) + 1 + dev_id)}'
-				self.db.execute("""
-					INSERT INTO virtual_machine_disks
-					(
-						virtual_machine_id,
-						disk_name,
-						disk_location,
-						disk_type,
-						disk_size,
-						image_file_name,
-						image_archive_name,
-						mount_disk
-					)
-					VALUES (
-						(select virtual_machines.id from virtual_machines inner join nodes on node_id=nodes.id where name=%s),
-						%s, %s, %s, %s, %s, %s, %s
-					)
-				""", (vm_host, disk_name, disk_loc, disk_type, disk_size, image, image_archive, mount_disk))
-				dev_id += 1
+			self.db.execute("""
+			INSERT INTO virtual_machine_disks
+				(
+					virtual_machine_id,
+					disk_name,
+					disk_location,
+					disk_type,
+					disk_size,
+					image_file_name,
+					mount_disk
+				)
+				VALUES (
+					(select virtual_machines.id from virtual_machines inner join nodes on node_id=nodes.id where name=%s),
+					%s, %s, %s, %s, %s, %s
+				)
+			""", (vm_host, disk_name, str(disk_loc), disk_type, disk_size, image, mount_disk))
+
+			# Increment the disk counter
+			dev_id += 1
```

Comment:
What is `dev_id` keeping track of? Look like its only read once and outside the loop.

Reply:
It's basically a counter to keep track of the next number for the new VM disk device name, so sda, sdb, sdc...etc. I'm taking the device name's ascii value and incrementing it by one to get the next device name.

Reply:
is this not doing that now?

https://github.com/Teradata/stacki/pull/725/files#diff-0b5a3b42e986ee3ddd7158c676f3bfe6R135

Reply:
It looks like that could be the case actually looking at it, I'll test adding disks to make sure the ordering is correct.